modules/olm-mirroring-catalog.adoc

```diff
@@ -0,0 +1,300 @@
+path of the database file.
++
+.Example output
+[source,terminal]
+----
+...
+image = registry.redhat.io/distributed-tracing/jaeger-all-in-one-rhel7@sha256:41f769c2c32f3f050aa42d86f084b739914ff9ba2f0aed2d9b0b69357b48459d
+operatorbundle_name = jaeger-operator.v1.17.6
+
+image = registry.redhat.io/distributed-tracing/jaeger-es-index-cleaner-rhel7@sha256:c64ac461d96523516a199bd132ad4d7148317e503a735028f0d8f7ba063a61cb
+operatorbundle_name = jaeger-operator.v1.17.6
+
+image = registry.redhat.io/distributed-tracing/jaeger-rhel7-operator:1.13.2
+operatorbundle_name = jaeger-operator.v1.13.2-1
+----
+
+... Use the results from the previous step to help you edit the `mapping.txt` file
+to only include the subset of images you want to mirror.
++
+For example, you can use the `image` values from the previous example output to
+find that the following matching lines exist in your `mapping.txt` file:
++
+.Matching image mappings in `mapping.txt`
+[source,txt]
+----
+...
+registry.redhat.io/distributed-tracing/jaeger-all-in-one-rhel7@sha256:41f769c2c32f3f050aa42d86f084b739914ff9ba2f0aed2d9b0b69357b48459d=quay.io/adellape/distributed-tracing-jaeger-all-in-one-rhel7:5cf7a033
+...
+registry.redhat.io/distributed-tracing/jaeger-es-index-cleaner-rhel7@sha256:c64ac461d96523516a199bd132ad4d7148317e503a735028f0d8f7ba063a61cb=quay.io/adellape/distributed-tracing-jaeger-es-index-cleaner-rhel7:ecfd2ca7
+...
+registry.redhat.io/distributed-tracing/jaeger-rhel7-operator:1.13.2=quay.io/adellape/distributed-tracing-jaeger-rhel7-operator:1.13.2
+...
+----
++
+In this example, if you only want to mirror these images, you would then remove
+all other entries in the `mapping.txt` file and leave only the above matching
+image mapping lines.
+
+.. Still on your workstation with unrestricted network access, use your modified
+`mapping.txt` file to mirror the images to your registry using the `oc image
+mirror` command:
++
+[source,terminal]
+----
+$ oc image mirror \
+    [-a ${REG_CREDS}] \
+    -f ./redhat-operator-index-manifests/mapping.txt
+----
+
+. Apply the ImageContentSourcePolicy:
++
+[source,terminal]
+----
+$ oc apply -f ./redhat-operator-index-manifests/imageContentSourcePolicy.yaml
+----
+
+. Mirror the Red Hat-provided index image from Quay.io to your registry:
++
+[source,terminal]
+----
+$ oc image mirror \
+    [-a ${REG_CREDS}] \
+    registry.redhat.io/redhat/redhat-operator-index:v4.6 \//<1>
+    <mirror_registry>:<port>/<namespace>/redhat-operator-index:v4.6 <2>
+----
+<1> Specify the index image for catalog that you mirrored content for in the
+previous step.
+<2> Specify where to mirror the index image.
+
+. Create a CatalogSource object that references your mirrored index image.
+
+.. Modify the following to your specifications and save it as a
+`catalogsource.yaml` file:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: my-operator-catalog
```

Comment:
this will work for a custom catalog.

but if you're mirroring a redhat default catalog (which are [defined here](https://github.com/operator-framework/operator-marketplace/tree/master/defaults)), we shouldn't need to do this - instead we can just add another ICSP that maps the mirrored index image to the same name referenced in the default catalogs.

so for example, if you've mirrored redhat operators to `<mirror_registry>:<port>/<namespace>/redhat-operator-index:v4.6`, you'd create an ICSP that lists `<mirror_registry>:<port>/<namespace>/redhat-operator-index:v4.6` as a mirror for `registry.redhat.io/redhat/redhat-operator-index:v4.6`

then you're done - you don't have to create a catalogsource object at all (because the default one will work)

Reply:
I guess in this scenario, the "Disabling the default OperatorHub sources" procedure earlier on would no longer be needed.

Reply:
After chatting with @kuiwang02, I think I'll leave the current "Disabling the default OperatorHub sources" + create-a-new-catalogsource version of this procedure intact for now (since it's been successfully QE'd) but look into altering this post-4.6 GA.